dashboard/app/batch_reproexport.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"net/http"
+
+	"google.golang.org/appengine/v2"
+	"google.golang.org/appengine/v2/log"
+)
+
+const exportTimeoutSeconds = 10000 * 2 // upstream has apx 7k reproducers, 1s each max (throttling)
+
+func handleBatchReproExport(w http.ResponseWriter, r *http.Request) {
+	ctx := appengine.NewContext(r)
+	reproExportPath := getConfig(ctx).ReproExportPath
+	if reproExportPath == "" {
+		return
+	}
+	if err := createScriptJob(ctx, "syzkaller", "export-repro",
+		exportReproScript(reproExportPath),
+		exportTimeoutSeconds, nil); err != nil {
+		log.Errorf(ctx, "createScriptJob: %s", err.Error())
+	}
+}
+
+func exportReproScript(archivePath string) string {
+	script := "\n" +
+		// "git clone --depth 1 --branch master --single-branch https://github.com/google/syzkaller\n" +
+		"git clone https://github.com/tarasmadan/syzkaller\n" +
+		"cd syzkaller\n" +
+		"git checkout syz_reprolist_by_namespace\n" +
+		"export CI=1\n" +
+		"./tools/syz-env \"" +
+		"go run ./tools/syz-reprolist/... -namespace upstream; " +
```

Comment:
I am not sure relying on syz-reprolist is a good idea.
syz-reprolist needs to know about changes in syzkaller code base:
https://github.com/google/syzkaller/blob/master/tools/syz-reprolist/reprolist.go#L172-L178
Nobody  updated that in the past 5 years (nor it's tested).

Reply:
I'd like to switch syz-reprolist from dashapi to jsonapi and once switched, delete the dashapi part.
Do you see the syz-reprolist future differently?

Reply:
I thought syz-reprolist works the way it worked before.
I am missing larger picture: what happens with the exported archive later? how do we export C repros where they are missing, if at all? What namespaces do we export from?

Reply:
> what happens with the exported archive later?

We'll need something like the build.sh and run.sh from https://github.com/dvyukov/syzkaller-repros.
Git is the alternative way to store these reproducers.
I decided to start from the GCS path because it costs 1 line now and allows me to experiment with the next steps.
P.S. Thanks for asking George about details.

> how do we export C repros where they are missing, if at all?

I don't understand the question. Could you please dive here a little?

> What namespaces do we export from?

I wanted to export the upstream reproducers first.

Reply:
It is a per-ns config now.